src/latte/core.clj

```diff
@@ -328,35 +330,50 @@ term `(type-of% term)` is replaced by the *type* of `term`."
       (do ;; (println "[parse-context-args] ctx=" ctx)
           ctx))))
 
-(defmacro term [& args]
-    (let [def-env {}
-          t (stx/parse def-env (last args))
-          ctx (parse-context-args def-env (butlast args))]
-      ;; (println "[term] t = " t " ctx = " ctx)
-      (if (latte.kernel.norm/beta-eq? def-env ctx t :kind)
-        '□
-        (let [ty (ty/type-of def-env ctx t)]
-          (list 'quote t)))))
+(defn apply
```

Comment:
maybe call it `apply-in-context`  or something like this... but don't forget I am
in the process of rewriting the kernel, then I will rewrite the `core` for clojure 1.9...
but I will accept your PR if it is focused on the  `apply` function, and of course if
it is renamed.